--- _specialsource_2_decompile.sh	2025-08-21 14:54:45.714863779 +0800
+++ _specialsource_2_decompile.sh	2025-08-21 15:00:30.748793793 +0800
@@ -4,8 +4,7 @@
 BASE_DIR="$1"
 WORKING_DIR="$1"
 
-wget_dir="${WORKING_DIR}/tmp"
-mkdir -p "${wget_dir}" || true
+wget_dir="${WORKING_DIR}/../tmp"
 
 decompilation_dir="${WORKING_DIR}/ss2/src"
 
@@ -33,19 +32,19 @@
   fi
 }
 
-ss2_script_download_file "fernflower.jar" "https://hub.spigotmc.org/stash/projects/SPIGOT/repos/builddata/raw/bin/fernflower.jar?at=838b40587fa7a68a130b75252959bc8a3481d94f" "29359771a5722c7f45ce723d567a86ae18cba7a72ee6ca7bd785235004aa1610"
-ss2_script_download_file "SpecialSource-2.jar" "https://github.com/hpfxd/PandaSpigot/raw/refs/heads/master/bin/SpecialSource-2.jar" "fc740dbb230c659df710084d0822b4a20d910bf707d6e66f935351bedf4e8bba"
+ss2_script_download_file "fernflower-ss.jar" "https://hub.spigotmc.org/stash/projects/SPIGOT/repos/builddata/raw/bin/fernflower.jar?at=838b40587fa7a68a130b75252959bc8a3481d94f" "29359771a5722c7f45ce723d567a86ae18cba7a72ee6ca7bd785235004aa1610"
+ss2_script_download_file "SpecialSource-2-Panda.jar" "https://github.com/hpfxd/PandaSpigot/raw/refs/heads/master/bin/SpecialSource-2.jar" "fc740dbb230c659df710084d0822b4a20d910bf707d6e66f935351bedf4e8bba"
 
 rm -rf "${decompilation_dir}"
 rm -rf "${decompilation_dir}/../target"
 mkdir -p "${decompilation_dir}"
 java -version
 set -x
-java -jar "${wget_dir}/fernflower.jar" -den=1 -dgs=1 -hdc=0 -rbr=0 -asc=1 -udv=0 "${wget_dir}/SpecialSource-2.jar" "${decompilation_dir}"
+java -jar "${wget_dir}/fernflower-ss.jar" -log=WARN -den=1 -dgs=1 -hdc=0 -rbr=0 -asc=1 -udv=0 "${wget_dir}/SpecialSource-2-Panda.jar" "${decompilation_dir}"
 set +x
 cd "${decompilation_dir}"
-jar xvf SpecialSource-2.jar
-rm -rf SpecialSource-2.jar
+jar xf SpecialSource-2-Panda.jar
+rm -rf SpecialSource-2-Panda.jar
 rm -rf com
 rm -rf joptsimple
 rm -rf META-INF
