From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NorthRealm <155140859+NorthRealm@users.noreply.github.com>
Date: Tue, 13 Jan 2026 19:28:31 +0800
Subject: [PATCH] Configure Gradle


diff --git a/build.gradle.kts b/build.gradle.kts
index 8c8aa2274fd97c9d3a2f24700f58e045e984f75d..954d2af8cdc60ab9f9588a081b5bca985ece215e 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,6 +1,8 @@
 plugins {
     id("pandaspigot.conventions")
     id("com.gradleup.shadow") version "9.0.0-beta7"
+    `maven-publish`
+    id("io.typst.gradlesource.spigot")
 }
 
 val minecraftVersion = "1_8_R3"
@@ -11,7 +13,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":pandaspigot-api"))
+    implementation(project(":colosseumspigot-api"))
 
     // PandaSpigot start - Configuration
     implementation("com.hpfxd.configurate:configurate-eo-yaml:1.0.0") {
@@ -50,21 +52,6 @@ dependencies {
     testImplementation("org.hamcrest:hamcrest-library:1.3")
 }
 
-fun TaskContainer.registerRunTask(
-    name: String, block: JavaExec.() -> Unit
-): TaskProvider<JavaExec> = register<JavaExec>(name) {
-    group = "pandaspigot"
-    standardInput = System.`in`
-    workingDir = rootProject.layout.projectDirectory.dir(
-        providers.gradleProperty("runWorkDir").orElse("run")
-    ).get().asFile
-
-    doFirst {
-        workingDir.mkdirs()
-    }
-    block(this)
-}
-
 tasks {
     shadowJar {
         mergeServiceFiles()
@@ -103,14 +90,14 @@ tasks {
         archiveClassifier.set("original")
         manifest {
             val git = Git(rootProject.layout.projectDirectory)
-            val gitHash = git("rev-parse", "--short=7", "HEAD").getText().trim()
-            val implementationVersion = System.getenv("GITHUB_RUN_NUMBER") ?: "\"$gitHash\""
+            val gitHash = git("rev-parse", "--short=9", "HEAD").getText().trim()
+            val implementationVersion = gitHash
             val date = git("show", "-s", "--format=%ci", gitHash).getText().trim()
             val gitBranch = git("rev-parse", "--abbrev-ref", "HEAD").getText().trim()
             attributes(
                 "Main-Class" to "org.bukkit.craftbukkit.Main",
                 "Implementation-Title" to "CraftBukkit",
-                "Implementation-Version" to "git-PandaSpigot-$implementationVersion",
+                "Implementation-Version" to "git-ColosseumSpigot-$implementationVersion",
                 "Implementation-Vendor" to date,
                 "Specification-Title" to "Bukkit",
                 "Specification-Version" to project.version,
@@ -122,24 +109,13 @@ tasks {
         }
     }
 
-    registerRunTask("runShadow") {
-        description = "Spin up a test server from the shadowJar archiveFile"
-        classpath(shadowJar.flatMap { it.archiveFile })
-    }
-
-    registerRunTask("runDev") {
-        description = "Spin up a non-shaded non-relocated test server"
-        classpath = java.sourceSets.main.get().runtimeClasspath
-        mainClass.set("org.bukkit.craftbukkit.Main")
-    }
-
-    register<RemapTask>("remap") {
+    register<io.typst.gradlesource.RemapTask>("remap") {
         description = "Remap the output JAR file using the deprecation mappings"
 
         val inTask = shadowJar.get()
         inJarFile.set(inTask.archiveFile)
         outJarFile.set(inTask.destinationDirectory.map { dir ->
-            val archiveName = ArchiveName.archiveNameFromTask(inTask).copy(classifier = "")
+            val archiveName = io.typst.gradlesource.archiveNameFromTask(inTask).copy(classifier = "")
             dir.file(archiveName.toFileName())
         })
 
@@ -151,3 +127,27 @@ tasks {
         dependsOn(named("remap"))
     }
 }
+
+publishing {
+    publications {
+        create<MavenPublication>("mavenJava") {
+            artifact(tasks.named("remap"))
+        }
+    }
+
+    System.getenv("COLOSSEUM_MAVEN_USERNAME").let { repoUsername ->
+        if (repoUsername == null) return@let // don't declare repository if username not declared
+
+        repositories {
+            maven {
+                name = "coffeewarehouse"
+                url = uri(System.getenv("COLOSSEUM_MAVEN_URL"))
+
+                credentials {
+                    username = repoUsername
+                    password = System.getenv("COLOSSEUM_MAVEN_PASSWORD") as String
+                }
+            }
+        }
+    }
+}
