From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NorthRealm <155140859+NorthRealm@users.noreply.github.com>
Date: Tue, 9 Sep 2025 11:20:07 +0800
Subject: [PATCH] Disable Nether and The End by default


diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index ea642606804c40b43bc447f295617b7c92f4a249..95ca915d4292b3f918452d9315d78a91451a9f23 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -413,7 +413,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
     }
 
     public boolean getAllowNether() {
-        return this.propertyManager.getBoolean("allow-nether", true);
+        return this.propertyManager.getBoolean("allow-nether", false);
     }
 
     public boolean getSpawnMonsters() {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 98a814601733dac73c9ad27447c13a3d35991f5d..6522920604da6fa859a55874000394011227e30e 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -562,7 +562,7 @@ public final class CraftServer implements Server {
 
     @Override
     public boolean getAllowNether() {
-        return this.getConfigBoolean("allow-nether", true);
+        return this.getConfigBoolean("allow-nether", false);
     }
 
     public boolean getWarnOnOverload() {
diff --git a/src/main/resources/configurations/bukkit.yml b/src/main/resources/configurations/bukkit.yml
index 8bbb835f4eeb4b188269de38060444a512eddc4a..e2844c20dc7dc970703c0ca6267490aeb44e6f90 100644
--- a/src/main/resources/configurations/bukkit.yml
+++ b/src/main/resources/configurations/bukkit.yml
@@ -13,7 +13,7 @@
 
 
 settings:
-    allow-end: true
+    allow-end: false
     warn-on-overload: true
     permissions-file: permissions.yml
     update-folder: update
