From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NorthRealm <155140859+NorthRealm@users.noreply.github.com>
Date: Tue, 9 Sep 2025 11:34:13 +0800
Subject: [PATCH] Set default Gamemode to Adventure


diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index a282523117c040d303e011ae77449dee68e7c446..d861937d0f3e3a02b2fd53f6c10e830ee161dcfe 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -162,7 +162,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
             this.setAllowFlight(this.propertyManager.getBoolean("allow-flight", false));
             this.setResourcePack(this.propertyManager.getString("resource-pack", ""), this.propertyManager.getString("resource-pack-hash", ""));
             this.setMotd(this.propertyManager.getString("motd", "A Minecraft Server"));
-            this.setForceGamemode(this.propertyManager.getBoolean("force-gamemode", false));
+            this.setForceGamemode(this.propertyManager.getBoolean("force-gamemode", true));
             this.setIdleTimeout(this.propertyManager.getInt("player-idle-timeout", 0));
             if (this.propertyManager.getInt("difficulty", 1) < 0) {
                 this.propertyManager.setProperty("difficulty", Integer.valueOf(0));
@@ -171,7 +171,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
             }
 
             this.generateStructures = this.propertyManager.getBoolean("generate-structures", true);
-            int i = this.propertyManager.getInt("gamemode", WorldSettings.EnumGamemode.SURVIVAL.getId());
+            int i = this.propertyManager.getInt("gamemode", WorldSettings.EnumGamemode.ADVENTURE.getId());
 
             this.r = WorldSettings.a(i);
             DedicatedServer.LOGGER.info("Default game type: " + this.r);
