From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NorthRealm <155140859+NorthRealm@users.noreply.github.com>
Date: Wed, 29 Oct 2025 11:23:15 +0800
Subject: [PATCH] Set settings.sample-count to 0 by default


diff --git a/src/main/java/net/minecraft/server/PacketStatusListener.java b/src/main/java/net/minecraft/server/PacketStatusListener.java
index c348744ed6ccda981bb811fe9180426946d6e3b5..2a9b543d59c73acabd9d45b779d6d0057f0a9cb9 100644
--- a/src/main/java/net/minecraft/server/PacketStatusListener.java
+++ b/src/main/java/net/minecraft/server/PacketStatusListener.java
@@ -110,11 +110,17 @@ public class PacketStatusListener implements PacketStatusInListener {
         }
 
         ServerPing.ServerPingPlayerSample playerSample = new ServerPing.ServerPingPlayerSample(event.getMaxPlayers(), profiles.size());
+        // ColosseumSpigot start
+        int samSub = Math.min( profiles.size(), org.spigotmc.SpigotConfig.playerSample );
+        if ( samSub <= 0 ) {
+            profiles.clear();
+        }
+        // ColosseumSpigot end
         // Spigot Start
         if ( !profiles.isEmpty() )
         {
             java.util.Collections.shuffle( profiles ); // This sucks, its inefficient but we have no simple way of doing it differently
-            profiles = profiles.subList( 0, Math.min( profiles.size(), org.spigotmc.SpigotConfig.playerSample ) ); // Cap the sample to n (or less) displayed players, ie: Vanilla behaviour
+            profiles = profiles.subList( 0, samSub ); // Cap the sample to n (or less) displayed players, ie: Vanilla behaviour
         }
         // Spigot End
         playerSample.a(profiles.toArray(new GameProfile[profiles.size()]));
diff --git a/src/main/java/org/spigotmc/SpigotConfig.java b/src/main/java/org/spigotmc/SpigotConfig.java
index ccc427c003212f6b3954f3a92c1b43d3db490b56..8e9d584eade9a4989b4e717d44d3cde2f9dad38d 100644
--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -328,7 +328,7 @@ public class SpigotConfig
     public static int playerSample;
     private static void playerSample()
     {
-        playerSample = getInt( "settings.sample-count", 12 );
+        playerSample = getInt( "settings.sample-count", 0 );
         System.out.println( "Server Ping Player Sample Count: " + playerSample );
     }
 
